tools/protoc-gen-gomex/mex.go

```diff
@@ -2504,13 +2815,20 @@ func (m *mex) generateMessage(file *generator.FileDescriptor, desc *generator.De
 			StreamOut:      gensupport.GetGenerateCudStreamout(message),
 			StringKeyField: gensupport.GetStringKeyField(message),
 		}
+		stateField := gensupport.FindField(message, "State")
+		if stateField != nil {
+			args.StateFieldType = m.support.GoType(m.gen, stateField)
+		}
 		m.cacheTemplate.Execute(m.gen.Buffer, args)
 		m.importUtil = true
 		m.importLog = true
 		if args.WaitForState != "" {
 			m.importErrors = true
 			m.importStrings = true
 		}
+		if args.ParentObjName != "" {
+			m.importSync = true
```

Comment:
I am thinking if overloading the meaning of `parent_obj_name` option might have implications in the future...  I am thinking if we will have some state data that's not explicitly coming from cloudlet, which is managed by CCRM/CRM. What do you think?

Reply:
I don't think I'm overloading it here. It's meant to identify the relationship between the "info" and its parent, i.e. CloudletInfo and its parent Cloudlet. And CloudletInfo is always what's used to send back information to the Controller in response to a Cloudlet change, which is the same thing the existing WaitFor code is doing, and the new code I added is doing (just over GRPC instead of notify). I agree "parent_obj_name" is not really an accurate name in terms of sending responses (although that's the primary purpose of the "info" structs). If anything I would rename it something like `response_obj_for`.